Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php

```diff
@@ -86,13 +86,25 @@ public function registerNodeChange(NodeInterface $node)
     }
 
     /**
-     * Fetches possible usages of the asset and registers nodes that use the asset as changed.
+     * Deprecated. Please use ContentCacheFlush::registerAssetChange
      *
+     * @deprecated
      * @param AssetInterface $asset
      * @return void
      */
     public function registerAssetResourceChange(AssetInterface $asset)
     {
+        $this->registerAssetChange($asset);
+    }
+
+    /**
+     * Fetches possible usages of the asset and registers nodes that use the asset as changed.
+     *
+     * @param AssetInterface $asset
+     * @return void
+     */
+    public function registerAssetChange(AssetInterface $asset)
+    {
         if (!$asset->isInUse()) {
```

Comment:
Well, the issue exists already, so no harm done either way. But my vote would indeed be for `Asset`, since that is what the method obviously expects. And the `AssetRepository` (which is what calls the emit method in `AssetService`, adding another very weird twist…) deals with `Asset`, so…